src/AutoRest.CSharp/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -0,0 +1,413 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Output.Models.Serialization.Json;
+using AutoRest.CSharp.Output.Models.Serialization.Xml;
+using Azure.Core;
+using AutoRest.CSharp.Utilities;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal static class RequestWriterHelpers
+    {
+        private const string PipelineVariable = "pipeline";
+        private const string PipelineField = "_" + PipelineVariable;
+
+        public static void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMethod, bool lowLevel)
+        {
+            using var methodScope = writer.AmbientScope();
+
+            var methodName = CreateRequestMethodName(clientMethod.Name);
+            var returnType = lowLevel ? typeof(Azure.Core.Request) : typeof(HttpMessage);
+            var visibility = lowLevel ? "public" : "internal";
```

Comment:
We should throw the visibility onto the RestClientMethod and handle this logic in LLCRestClient